test/extended/cli/version.go

```diff
@@ -0,0 +1,76 @@
+package cli
+
+import (
+	"fmt"
+	"regexp"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	"strings"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[sig-cli] oc version", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("oc-version")
+	ocAdmin := oc.AsAdmin()
+
+	g.It("check kubernetes version", func() {
+		podNames, err := ocAdmin.Run("get").Args("pods", "-l", "app=kube-controller-manager", "-n", "openshift-kube-controller-manager", "-o=jsonpath={.items..metadata.name}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		podList := strings.Split(podNames, " ")
+
+		imageName, err := ocAdmin.Run("get").Args("pod", "-n", "openshift-kube-controller-manager", podList[0], "-o=jsonpath='{.spec.containers[0].image}'").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		masterNode, err := exutil.GetFirstMasterNode(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		imageInfo, err := exutil.DebugNodeWithChroot(oc, masterNode, "oc", "image", "info", "--registry-config=/var/lib/kubelet/config.json", strings.Trim(imageName, "'"))
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		re1 := regexp.MustCompile(`io.openshift.build.versions=kubernetes=[1-9]{1}.[1-9]{1,2}.[1-9]{1}`)
+		kversion := re1.FindAllString(imageInfo, -1)
+		kversion = strings.Split(kversion[0], "=")
+		kversionNum := kversion[len(kversion)-1]
+
+		version, err := ocAdmin.Run("version").Output()
+		o.Expect(version).To(o.ContainSubstring(kversionNum))
+		if err != nil {
+			e2e.Failf("Fail to get the Kubernetes version")
+		}
+		re := regexp.MustCompile(`Kubernetes Version: v[0-9]+\.[0-9]+\.[0-9]+($|\+)`)
+		if re == nil {
+			e2e.Failf("version regexp err!")
+		}
+		result := re.FindAllStringSubmatch(version, -1)
+		if match, _ := regexp.MatchString(kversionNum, result[0][0]); !match {
+			e2e.Failf("Failed to match Kuberntes version %v & OC Version %v", kversionNum, result[0][0])
+		}
+
+		nodeName, err := ocAdmin.WithoutNamespace().Run("get").Args("nodes", "--selector=node-role.kubernetes.io/worker=", "-o=jsonpath={.items[*].metadata.name}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		node := strings.Fields(nodeName)
+
+		for _, v := range node {
+			kubeletVersion, err := ocAdmin.WithoutNamespace().Run("get").Args("node", fmt.Sprintf("%s", v), "-o", "yaml", "-o", "custom-columns=VERSION:.status.nodeInfo.kubeletVersion").Output()
```

Comment:
You can replace this with a single `oc get no -ocustom-columns=VERSION:.status.nodeInfo.kubeletVersion` and then just iterate over result ignoring the first row.

Reply:
Done